maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/FeatureDecisionService.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.justice.laa.crime.orchestration.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.justice.laa.crime.orchestration.dto.WorkflowRequest;
+import uk.gov.justice.laa.crime.orchestration.dto.validation.UserSummaryDTO;
+import uk.gov.justice.laa.crime.orchestration.enums.CurrentFeatureToggles;
+import uk.gov.justice.laa.crime.orchestration.enums.FeatureToggleAction;
+
+@Slf4j
+@Service
+public class FeatureDecisionService {
+
+    private final MaatCourtDataService maatCourtDataService;
+
+    public FeatureDecisionService(final MaatCourtDataService maatCourtDataService) {
+        this.maatCourtDataService = maatCourtDataService;
+    }
+
+    /**
+     * Returns whether the CCC Service is enabled based on whether the user has the requisite
+     * feature flag (CurrentFeatureToggles.CALCULATE_CONTRIBUTION) and action (Create) in their
+     * user summary information.
+     * @param workflowRequest The workflow request to check against.
+     * @return A boolean describing whether C3 should be invoked for the user's request.
+     */
+    public boolean isC3Enabled(WorkflowRequest workflowRequest) {
+        UserSummaryDTO userSummaryDTO = maatCourtDataService.getUserSummary(
+            workflowRequest.getUserDTO().getUserName());
```

Comment:
I think the issue with having a generic method is that the consumer then needs to know what constitutes a feature being enabled. Sure at the moment it might just be the presence of the row in the FEATURE_TOGGLE table, but if this changes in the future you're then in a situation where logic becomes duplicated and inconsistent. For what it's worth, I did a bit of reading around on the subject before writing this PR and settled on the [approach described by Martin Fowler on his site regarding feature toggles](https://martinfowler.com/articles/feature-toggles.html). I think it also makes testing easier and more comprehensive as you're then testing specific features rather than a generic "does feature exist" type of logic.

Not sure on consolidating with the existing call, will have to have a look what's happening there first.